massdriver/resource_available_cidr.go

```diff
@@ -0,0 +1,110 @@
+package massdriver
+
+import (
+	"context"
+	"net"
+
+	"github.com/massdriver-cloud/cola/pkg/cidr"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAvailableCidr() *schema.Resource {
+	return &schema.Resource{
+		Description: "A Massdriver artifact for exporting a connectable type",
+
+		CreateContext: resourceAvailableCidrCreate,
+		ReadContext:   schema.NoopContext,
+		UpdateContext: schema.NoopContext, //resourceArtifactUpdate,
+		DeleteContext: resourceAvailableCidrDelete,
+
+		Schema: map[string]*schema.Schema{
+			"parent_cidrs": {
```

Comment:
What do you think about  `find_from` / `find_in`? I think `parent` here reflects the data structure more than something that's easier for users of this resource to understand.  I think the description is really clear if people wanted to know more.

```hcl
resource "a" "b" {
  find_from = ["10.0.0.0/20"]
  used_cidrs = [....]
}
```

Reply:
I want to keep the word `cidr` in the field name so people understand what it is. We can drop `parent`, but people should understand this is a list of 1 or more CIDRs.

Reply:
How do you get a range from multiple "parent" CIDRs?

Reply:
> How do you get a range from multiple "parent" CIDRs?

In the case of VPCs, you can select from any of the private ranges (`10.0.0.0/8`, `172.16.0.0/20`, `192.168.0.0/16`).
In the case of subnets, I believe all the cloud networks (AWS VPC, GCP Global Network and Azure VNet) allow the network to encompass multiple ranges.

Reply:
I can get that for the VPC but I don’t know that anybody would use it that way. Do you really not care which class you end up in?